sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/Counter.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.beam.sdk.metrics;
+
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
+
+/**
+ * A metric that reports a single long value and can be incremented or decremented.
+ */
+@Experimental(Kind.METRICS)
+public class Counter {
+
+  private final MetricName name;
+
+  Counter(MetricName name) {
+    this.name = name;
+  }
+
+  /** Increment the counter. */
+  public void inc() {
+    inc(1);
+  }
+
+  /** Increment the counter by the given amount. */
+  public void inc(long n) {
```

Comment:
Went with these names to mimic Dropwizard: http://metrics.dropwizard.io/3.1.0/apidocs/com/codahale/metrics/Counter.html.

Although by that logic we should consider renaming `Distribution` to http://metrics.dropwizard.io/3.1.0/apidocs/com/codahale/metrics/Histogram.html

How strongly do you think it should be add?